.travis.yml

```diff
@@ -32,75 +32,86 @@ addons:
-
-# Install miniconda
-before_install:
-    - source continuous-integration/travis/install_$TRAVIS_OS_NAME.sh
+      - env: JOB="Py2Doctest" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -b doctest'
+             CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+      - env: JOB="Py2Coverage" PYTHON_VERSION=2.7 SETUP_CMD='test --online --coverage'
+             CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 PIP_DEPENDENCIES='pytest-cov'
 
 install:
-    # Add a conda env and set up extra binstar channels
-    - conda create --yes -n test python=$PYTHON_VERSION
-    - conda config --add channels sunpy --add channels astropy --add channels astropy-ci-extras --add channels https://conda.binstar.org/sunpy/channel/citesting
-    - source activate test
-    # Install default dependencies
-    - conda install --yes pip
-    - conda install -q --yes numpy$NUMPY_VERSION pandas$PANDAS_VERSION scipy matplotlib$MATPLOTLIB_VERSION requests beautiful-soup sqlalchemy scikit-image pytest jinja2 cython wcsaxes
-    - pip install pytest-cov coverage coveralls
-    # Install sunpy conda packages
-    - conda install -q --yes numpy=$NUMPY_VERSION suds-jurko glymur
-    # Install the correct version of astropy
-    - if [[ $TEST_MODE != 'astropy-dev' ]]; then conda install -q --yes astropy>=0.4.0 numpy$NUMPY_VERSION; fi
-    - if [[ $TEST_MODE == 'astropy-dev' ]]; then pip install git+git://github.com/astropy/astropy.git; fi
-    # Install sphinx if needed
-    - if [[ $TEST_MODE == 'sphinx' || $TEST_MODE == 'doctest' ]]; then conda install -q --yes sphinx numpy$NUMPY_VERSION; fi
-    - if [[ $TEST_MODE == 'sphinx' || $TEST_MODE == 'doctest' ]]; then python -c "import sunpy.data; sunpy.data.download_sample_data()"; fi
+    - echo $CONDA_DEPENDENCIES
+    - git clone git://github.com/astropy/ci-helpers.git
+    - source ci-helpers/travis/setup_conda_$TRAVIS_OS_NAME.sh
+    - source continuous-integration/travis/openjpeg.sh # so glymur knows where to look
 
-script:
-    - if [[ $TEST_MODE == 'offline' || $TEST_MODE == 'astropy-dev' ]]; then python setup.py test; fi
-    - if [[ $TEST_MODE == 'figure' ]]; then python setup.py test --figure; fi
-    - if [[ $TEST_MODE == 'sphinx' ]]; then python setup.py build_sphinx -w; fi
-    - if [[ $TEST_MODE == 'online' ]]; then python setup.py test --online --coverage --cov-report=html; fi
-    #- if [[ $TEST_MODE == 'skip' ]]; then coveralls --rcfile='./sunpy/tests/coveragerc'; fi
-    - if [[ $TEST_MODE == 'doctest' ]]; then python setup.py build_sphinx -b doctest; fi
-    - if [[ $TEST_MODE == 'python3' ]]; then continuous-integration/python3.sh; fi
+    # Download the sample data for the build of the documentation.
+    - if [[ $SETUP_CMD == *sphinx* ]]; then python -c "import sunpy.data; sunpy.data.download_sample_data()"; fi
+    - if [[ $SETUP_CMD == *figure* ]]; then wget http://raw.githubusercontent.com/dpshelio/sunpy-figure-tests/np1104/conda_env; conda create --name testFigure --file conda_env ; source activate testFigure; fi
```

Comment:
@Cadair, you commented on this, but it updated and your comment got hidden. 
the plan is to update this once sunpy/sunpy-figure-tests#5 gets merged.  This way we can test the figures with the packages used to build them.